app/controllers/workflow_service_controller.rb

```diff
@@ -0,0 +1,80 @@
+class WorkflowServiceController < ApplicationController
+  ##
+  # Is a document closeable?
+  def closeable
+    @status = check_if_can_close_version
+    render json: @status
+  end
+
+  ##
+  # Is a document openable?
+  def openable
+    @status = check_if_can_open_verison
+    render json: @status
+  end
+
+  ##
+  # Has an object been published?
+  def published
+    @status = check_if_published
+    render json: @status
+  end
+
+  ##
+  # Has an object been submitted?
+  def submitted
+    @status = check_if_submitted
+    render json: @status
+  end
+
+  ##
+  # Has an object been accessioned?
+  def accessioned
+    @status = check_if_accessioned
+    render json: @status
+  end
+
+  private
+
+  ##
+  # Ported over logic from app/helpers/dor_object_helper.rb#LN119
+  # @return [Boolean]
+  def check_if_published
+    return true if Dor::WorkflowService.get_lifecycle('dor', params[:pid], 'published')
```

Comment:
Yes, `lib` is fine with me. caching active workflows may not be effective as they may change at anytime (although they do send JMS messages on changes which we could consume for invalidation). workflows that are reliably cacheable are those that have been archived which is a large percentage of them.